crates/biome_js_type_info/src/local_inference.rs

```diff
@@ -678,6 +684,40 @@ impl Type {
         .into()
     }
 
+    pub fn from_js_binary_expression(expr: &JsBinaryExpression) -> Self {
+        let (Ok(left), Ok(operator), Ok(right)) = (expr.left(), expr.operator(), expr.right())
+        else {
+            return Self::unknown();
+        };
+
+        let left = Self::from_any_js_expression(&left);
+        let right = Self::from_any_js_expression(&right);
+
+        match operator {
+            JsBinaryOperator::StrictEquality => match (left.inner_type(), right.inner_type()) {
+                (TypeInner::Literal(left), TypeInner::Literal(right)) => {
+                    if left == right {
+                        Literal::Boolean(true.into()).into()
+                    } else {
+                        Literal::Boolean(false.into()).into()
+                    }
+                }
+                _ => Self::boolean(),
```

Comment:
Not necessary for this PR, but ideally we could even do something like `Self::typeof_expression(Expression::StrictEquality(left, right))`, so that we can do more advanced inference even when comparing variables that we know the type of, for instance.

I'd leave it for a future improvement ;)

Reply:
Yeah it might be wanted in some day. I'd keep it simple until it is needed by some rules.